pkg/cmd/repo/edit/edit.go

```diff
@@ -181,6 +184,7 @@ func editRun(ctx context.Context, opts *EditOptions) error {
 			"hasIssuesEnabled",
 			"hasProjectsEnabled",
 			"hasWikiEnabled",
+			"hasDiscussionsEnabled",
```

Comment:
We cannot unconditionally query the `hasDiscussionsEnabled` GraphQL field because it's a fairly recent API that does not exist on older GitHub Enterprise versions.

There is some prior art to conditionally querying GraphQL fields, but I'd say that for now it's the easiest to avoid querying this value and instead default to `false` in interactive flow (which is the only place that otherwise uses this value, I think?)

Reply:
I will comment out it.

```diff
diff --git a/pkg/cmd/repo/edit/edit.go b/pkg/cmd/repo/edit/edit.go
index 01bdead1..ae670bcf 100644
--- a/pkg/cmd/repo/edit/edit.go
+++ b/pkg/cmd/repo/edit/edit.go
@@ -184,7 +184,8 @@ func editRun(ctx context.Context, opts *EditOptions) error {
 			"hasIssuesEnabled",
 			"hasProjectsEnabled",
 			"hasWikiEnabled",
-			"hasDiscussionsEnabled",
+			// TODO: GitHub Enterprise Server does not support has_discussions yet
+			// "hasDiscussionsEnabled",
 			"homepageUrl",
 			"isInOrganization",
 			"isTemplate",
@@ -279,7 +280,8 @@ func interactiveChoice(r *api.Repository) ([]string, error) {
 		optionIssues,
 		optionMergeOptions,
 		optionProjects,
-		optionDiscussions,
+		// TODO: GitHub Enterprise Server does not support has_discussions yet
+		// optionDiscussions,
 		optionTemplateRepo,
 		optionTopics,
 		optionVisibility,
```

Should I hide "[ ] Discussions" too?

Before

<img width="654" alt="image" src="https://user-images.githubusercontent.com/19714/214171007-cdf5ee67-4da1-4e61-9558-0da96dfec801.png">

After
<img width="610" alt="image" src="https://user-images.githubusercontent.com/19714/214170765-72d6352c-5c7f-4611-b8a4-b716adccd022.png">

Reply:
https://github.com/cli/cli/pull/6903/commits/0877f56e7449544781fa88f9b6c9b85ebc88f355